roles/inspec/tasks/main.yml

```diff
@@ -10,18 +10,71 @@
     mode: 0755
   with_items:
     - /etc/inspec
-    - /etc/inspec/controls
-    - /etc/inspec/libraries
+    - /etc/inspec/host-controls
+    - /etc/inspec/host-controls/controls
+    - /etc/inspec/host-controls/libraries
 
-- name: inspec config file
+- name: inspec dependency profile directories
+  file:
+    dest: '{{ inspec.dependency_path }}/archive/'
+    mode: 0766
+    state: directory
+  with_items: '{{inspec.profile_dependencies }}'
+  when: inspec.dependency_install_method == 'tar'
+
+- name: create current directory for inspec dependencies
+  file:
+    dest: '{{ inspec.dependency_path }}/current/'
+    mode: 0766
+    state: directory
+  when: inspec.dependency_install_method == 'tar'
+
+- name: fetch inspec dependencies when install method is tar
+  get_url:
+    url: '{{ item.url }}'
+    dest: '{{ inspec.dependency_path }}/archive/{{ item.name }}-{{ item.version }}'
+    mode: 0644
+  with_items: '{{ inspec.profile_dependencies }}'
+  when: inspec.dependency_install_method == 'tar'
+
+- name: untar inspec dependencies
+  unarchive:
+    src: '{{ inspec.dependency_path }}/archive/{{ item.name }}-{{ item.version }}'
+    dest: '{{ inspec.dependency_path }}'
+    copy: no
+  with_items: '{{inspec.profile_dependencies }}'
+  when: inspec.dependency_install_method == 'tar'
+
+- name: create inspec dependency current symlink
+  file:
+    src: '{{ inspec.dependency_path }}/{{ item.name }}-{{ item.version }}'
+    dest: '{{ inspec.dependency_path }}/current/{{ item.name }}-{{ item.version }}'
```

Comment:
the idea with the `current` link is that you don't need to specify the version in it as its already known.   `dest: '{{ inspec.dependency_path }}/current/{{ item.name }}'`